protocol/x/clob/keeper/final_settlement.go

```diff
@@ -29,11 +30,14 @@ func (k Keeper) mustCancelStatefulOrdersForFinalSettlement(ctx sdk.Context, clob
 	// rather than halt the chain.
 	removeStatefulOrderWithoutPanicing := func(ctx sdk.Context, orderId types.OrderId) {
 		defer func() {
-			if r := recover(); r != nil {
-				k.Logger(ctx).Error(
-					"mustCancelStatefulOrdersForFinalSettlement: Failed to remove stateful order with OrderId %+v: %v",
+			if err := recover(); err != nil {
+				log.ErrorLog(
+					ctx,
+					"mustCancelStatefulOrdersForFinalSettlement: Failed to remove stateful order",
+					"orderId",
 					orderId,
-					r,
+					"error",
+					err,
 				)
 			}
 		}()
```

Comment:
unrelated to this PR - but seems like we do this a lot in many places. don't have numbers to back me up but these can be costly if they are in hot code paths. 

created [this](https://linear.app/dydx/issue/CLOB-1065/see-if-recovering-from-panics-in-defer-func-is-costly-and-potentially) ticket to do some analysis and audit panics.